### ISSUE TITLE
Move booking page to SSG + ISR instead of SSR

### ISSUE THREAD

**Issue by agustif** [calcom/cal.com#2071]
### Is your proposal related to a problem?
Improvement of user/book page, after seeing #2057 and thinking about how we could improve it further.

What if we move the user/book page to ISR, and call whenever the user profile/ event types are modified and thus the page needs to be updated?

https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration

**Comment by hariombalhara** [comment]
ISR can totally improve the User experience and PSI score.But it would add some delay(in multiple seconds) for any changes in app to be reflected in the page.
 e.g. I disable an event and it would still show up for a few seconds. 

If I keep the revalidation too low(say 1 second), it would be like SSR but would have the benefit of lower CPU cost. But there would be one big limitation then which is not able to serve authenticated content which is possible right now(as it's SSR)

**Example use case -**
If I am authenticated with cal.com and I want to book a meeting with a team that allows team only bookings(Hypothetical), I should be served a message that I am not part of the team.
Even though this can be done with ISR but it would mean another request from the browser to identify this use case and this additional delay in showing the useful info. 

To summarize, as a user I might want a super fast but also very personalized experience on booking page. 

I believe the call should be taken on what features we have in roadmap for booking pages.

**Comment by PeerRich** [comment]
@KATT had some opinions on this. i mostly agree with @hariombalhara

**Comment by PeerRich** [comment]
also, @emrysal is working on cached calendar availability which may open up new avenues for us in terms of rendering
